File: shopapp-backend/src/main/java/com/project/shopapp/dtos/OrderDTO.java
package com.project.shopapp.dtos;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import lombok.*;

import java.time.LocalDate;
import java.util.Date;
import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class OrderDTO {
    @JsonProperty("user_id")
    @Min(value = 1,message = "Userid must be >0")
    private Long userId;

    @JsonProperty("fullname")
    private String fullName;

    private String email;

    @JsonProperty("phone_number")
    @NotBlank(message = "phoneNumber ko đc để trống")
    @Size(min = 5,message = "Phone number tối thiểu 5 kí tự")
    private String phoneNumber;

    @NotBlank(message = "Địa chỉ ko đc rỗng")
    @JsonProperty("address")
    private String address;

    private String note;
    private String status;
    @JsonProperty("total_money")
    @Min(value=0,message = "Total money must be >0")
    private float totalMoney;

    @JsonProperty("shipping_date")
    private LocalDate shippingDate;
    @JsonProperty("shipping_method")
    private String shippingMethod;

    @JsonProperty("shipping_address")
    private String shippingAddress;

    @JsonProperty("payment_method")
    private String paymentMethod;
    @JsonProperty("cart_items")
    private List<CartItemDTO> cartItemDTOS;
}

File: shopapp-angular-V16/src/app/dtos/order/order.dto.ts
import { IsString, 
  IsNotEmpty, 
  IsPhoneNumber, 
  IsNumber, ArrayMinSize, 
  ValidateNested, 
  Length 
} from 'class-validator';
import { Type } from 'class-transformer';
import { CartItemDTO } from './cart.item.dto';
import { OrderDetail } from 'src/app/models/order.detail';

export class OrderDTO {
  user_id: number;

  fullname: string;

  email: string;

  phone_number: string;
  
  address: string;
  note: string;
  order_date: Date;
  total_money: number;
  
  shipping_method: string;
  
  payment_method: string;
    shipping_address: string;
    shipping_date: Date; // Dạng chuỗi ISO 8601
  coupon_code: string;
  
  cart_items: { product_id: number, quantity: number }[]; // Thêm cart_items để lưu thông tin giỏ hàng
  status:string;
  order_details: OrderDetail[];
  constructor(data: any) {
    this.user_id = data.user_id;
    this.fullname = data.fullname;
    this.email = data.email;
    this.phone_number = data.phone_number;
    this.address = data.address;
    this.note = data.note;
    this.order_date = data.order_date;
    this.total_money = data.total_money;
    this.shipping_method = data.shipping_method;
    this.payment_method = data.payment_method;
    this.shipping_address = data.shipping_address;
    this.shipping_date = data.shipping_date;
    this.coupon_code = data.coupon_code;
    this.cart_items = data.cart_items;
    this.status=data.status;
    this.order_details=data.order_details;
  }
}

